### DISCUSSION TITLE
https://auth.tesla.com/oauth2/v3/token -> 400 Token refresh failed

### DISCUSSION THREAD

**Discussion by Sasquatch1976** [teslamate-org/teslamate#1486] [Q&A]
Hi. Recently my Teslamate installation (Google Cloud hosted) "broke". I did update to newest version and all.

When running docker-compose up (verbose og simply looging in logs) I'm faced with the below shown sniplet (and alot more of cause)

Did Tesla recently (yet again) change the MFA or auth verification.

I can log into my Tesla account with no errors and have tested with a Tesla password change.

teslamate_1  | waiting for postgres at database:5432
teslamate_1  | waiting for postgres at database:5432
teslamate_1  | waiting for postgres at database:5432
teslamate_1  | 2021-03-28 16:54:38.365 [info] Migrations already up
teslamate_1  | 2021-03-28 16:54:41.805 [info] Version: 1.22.0
teslamate_1  | 2021-03-28 16:54:42.404 [error] POST https://auth.tesla.com/oauth2/v3/token -> 400 (394.858 ms)
teslamate_1  | 2021-03-28 16:54:42.406 [warn] Token refresh failed: %TeslaApi.Error{
teslamate_1  |   env: %Tesla.Env{
teslamate_1  |     __client__: %Tesla.Client{adapter: nil, fun: nil, post: [], pre: []},
teslamate_1  |     __module__: TeslaApi.Auth,
teslamate_1  |     body: %{
teslamate_1  |       "error" => "server_error",
teslamate_1  |       "error_description" => "Internal server error",
teslamate_1  |       "error_uri" => "https://auth.tesla.com/error/reference/58d9980e-71d6-4496-9516-fa3580243848-1616943282361"
teslamate_1  |     },
teslamate_1  |     headers: [
teslamate_1  |       {"server", "nginx"},
teslamate_1  |       {"content-type", "application/json"},
teslamate_1  |       {"x-dns-prefetch-control", "off"},
teslamate_1  |       {"x-frame-options", "DENY"},
teslamate_1  |       {"strict-transport-security", "max-age=15552000; includeSubDomains"},
teslamate_1  |       {"x-download-options", "noopen"},
teslamate_1  |       {"x-content-type-options", "nosniff"},
teslamate_1  |       {"x-xss-protection", "1; mode=block"},
teslamate_1  |       {"x-request-id", "58d9980e-71d6-4496-9516-fa3580243848"},
teslamate_1  |       {"x-correlation-id", "58d9980e-71d6-4496-9516-fa3580243848"},
teslamate_1  |       {"access-control-allow-origin", "*"},
teslamate_1  |       {"x-response-time", "2.204ms"},
teslamate_1  |       {"date", "Sun, 28 Mar 2021 14:54:42 GMT"},
teslamate_1  |       {"connection", "close"}
teslamate_1  |     ],
teslamate_1  |     method: :post,
teslamate_1  |     opts: [],
teslamate_1  |     query: [],
teslamate_1  |     status: 400,
teslamate_1  |     url: "https://auth.tesla.com/oauth2/v3/token"
teslamate_1  |   },
teslamate_1  |   message: nil,
teslamate_1  |   reason: :token_refresh
teslamate_1  | }
teslamate_1  | 2021-03-28 16:54:42.406 [info] Scheduling token refresh in 1 h
teslamate_1  | 2021-03-28 16:54:42.416 [info] Running TeslaMateWeb.Endpoint with cowboy 2.8.0 at :::4000 (http)
teslamate_1  | 2021-03-28 16:54:42.439 [info] Access TeslaMateWeb.Endpoint at http://teslamate.holmerjensen.dk
teslamate_1  | 2021-03-28 16:54:42.622 [info] GET https://owner-api.teslamotors.com/api/1/vehicles -> 401 (178.556 ms)
teslamate_1  | 2021-03-28 16:54:42.766 [warn] Using fallback vehicles:

**Comment by adriankumpf** [comment]
Can you try signing out from the settings page and then signing in again? Alternatively, you can enter the tokens manually.

**Reply by Sasquatch1976** [reply]
I get a 404 page not found. No GUI available.... (Or is there a direct link perhaps)? I have no way of signing in... I an not familiar with manually entering tokens. Is there a guide/manual perhaps I can follow ? :-) Thanks in advance.

**Reply by Sasquatch1976** [reply]
I dont know if this is of any help Adrian :-)

teslamate_1  | 2021-04-02 18:46:25.983 [info] Starting logger for 'Mystique'
teslamate_1  | 2021-04-02 18:46:26.244 [error] Error / unauthorized
mosquitto_1  | 1617381986: New connection from 172.20.0.6 on port 1883.
mosquitto_1  | 1617381986: New client connected from 172.20.0.6 as TESLAMATE_302E3830313 (p2, c1, k60).
teslamate_1  | 2021-04-02 18:46:26.372 [info] MQTT connection has been established
teslamate_1  | 2021-04-02 18:46:26.688 [info] tzdata release in place is from a file last modified Tue, 22 Dec 2020 23:35:21 GMT. Release file on server was last modified Sun, 24 Jan 2021 19:35:23 GMT.
teslamate_1  | 2021-04-02 18:46:28.504 [info] Tzdata has updated the release from 2020e to 2021a
teslamate_1  | 2021-04-02 18:47:23.164 [info]     :alarm_handler: {:set, {TeslaMate.Vehicles.Vehicle_1_api_error, :fuse_blown}}

...

     | time="2021-04-02T16:47:27Z" level=error msg="Router defined multiple times with different configurations in [grafana-allan-jensen-ef7e664ac296e5545fc98a4e735ed73eb4a2310233bc18a8a277d2828f2c2b59 grafana-teslamate-ff05ccaf0b4b46a6878365599ceb8c5f9d5f788d7f61b2c8ee73c82b0bb4d04b]" providerName=docker routerName=grafana-insecure
proxy_1      | time="2021-04-02T16:47:27Z" level=error msg="Router defined multiple times with different configurations in [grafana-allan-jensen-ef7e664ac296e5545fc98a4e735ed73eb4a2310233bc18a8a277d2828f2c2b59 grafana-teslamate-ff05ccaf0b4b46a6878365599ceb8c5f9d5f788d7f61b2c8ee73c82b0bb4d04b]" routerName=grafana providerName=docker
proxy_1      | time="2021-04-02T16:47:27Z" level=error msg="Router defined multiple times with different configurations in [teslamate-allan-jensen-8dc590ee1ff717af55175ddc3ebc085443b5cd68380e82e80ea5f420d1257829 teslamate-teslamate-d9cf4e00282507011140ef04677f33e6dbdcef6fe71295c326cd16bd6202f8e5]" providerName=docker routerName=teslamate-insecure
proxy_1      | time="2021-04-02T16:47:27Z" level=error msg="Router defined multiple times with different configurations in [teslamate-allan-jensen-8dc590ee1ff717af55175ddc3ebc085443b5cd68380e82e80ea5f420d1257829 teslamate-teslamate-d9cf4e00282507011140ef04677f33e6dbdcef6fe71295c326cd16bd6202f8e5]" providerName=docker routerName=teslamate-ws
